### PR TITLE
feat(export-to-language) implement export to language for ruby RUBY-2873

### PR DESCRIPTION
**Pull request by Neilshweky**
This PR is for adding ruby as an output language. I left a bunch of comments on things I wasn't sure about.

### CODE REVIEW

**Review by addaleax — APPROVED** [review]
This looks good from the Compass side of things -- is there anything in the ruby templates/etc. itself that you think requires special attention from us? I don't know if any of us are really going to be helpful when it comes to reviewing ruby code or the driver API usage.

**Comment by Neilshweky** [comment]
Hey @addaleax thanks so much for reviewing! I think some of the things you can help with are some of the comments I left which are more conceptual and have to do with other languages.

**Comment by Neilshweky** [comment]
@addaleax I also just added Ruby to the export to language select in the compass UI, if you want to check that out (and rebased to get rid of the merge conflicts)

File: packages/bson-transpilers/test/yaml/bson-methods.yaml [hunk 1]
```diff
@@ -92,6 +100,12 @@ tests:
             csharp: new MongoDBRef("coll", new ObjectId(), "db").DatabaseName
             shell: new DBRef('coll', new ObjectId(), "db").getDb()
             object: "'db'"
+            ruby: |-
+                BSON::DBRef.new({
+                  '$ref' => 'coll',
+                  '$id' => BSON::ObjectId.new,
+                  '$db' => 'db'
+                }).database
```
Comment (p-mongo) [inline packages/bson-transpilers/test/yaml/bson-methods.yaml:108]: The curly braces (`{}`) are redundant here, though this syntax does work.

File: packages/bson-transpilers/test/yaml/bson-methods.yaml [hunk 2]
```diff
@@ -543,6 +630,7 @@ tests:
           shell: ObjectId.fromDate(new Date('Thu, 03 Feb 1972 00:00:00 GMT'))
           # TODO: fix? this previously called ObjectId.createFromTime with a Date, but that function does not actually take a Date argument
           # object: new ObjectId('03ede8800000000000000000')
+          ruby: BSON::ObjectId.from_time(Time.new(1972, 2, 3, 0, 0, 0).utc)
```
Comment (p-mongo) [inline packages/bson-transpilers/test/yaml/bson-methods.yaml:633]: There's a `Time.utc` method if you need the UTC time.

**Review by p-mongo — APPROVED** [review]
:+1:  The Ruby output looks good to me.

**Pull request merged**